robot_smach_states/src/robot_smach_states/navigation/constraint_functions/arms_reach_constraints.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import absolute_import
+
+# System
+import math
+
+# ROS
+import rospy
+from geometry_msgs.msg import Point
+
+# TU/e Robotics
+from cb_planner_msgs_srvs.msg import OrientationConstraint, PositionConstraint
+
+
+def arms_reach_constraint(pose_designator, arm_designator, look=True):
+    """
+    Position so that the arm can reach the position/entity
+    :param pose_designator: designator that resolves to a FrameStamped
+    :param arm_designator: which arm to use for manipulation
+    :param look: whether or not the orientation must be constrained as well
+    :return: tuple(PositionConstraint, OrientationConstraint)
+    """
+    arm = arm_designator.resolve()
+    if not arm:
+        rospy.logerr("Could not resolve arm")
+        return None
+
+    radius = math.hypot(arm.base_offset.x(), arm.base_offset.y())
+
+    pose = pose_designator.resolve()
+    if not pose:
+        rospy.logerr("No such place_pose")
```

Comment:
Plz log which designator failed, in each of them. You should be able to just `str()` them

Reply:
Currently this is a proof of concept. I want to decide which approach is better, passing functions or using designators. After that decision is made I will further develop either this PR or #1006.